docs/ai/conceptual/prompt-engineering-in-dot-net.md

```diff
@@ -0,0 +1,190 @@
+
+Directives are more flexible than imperatives:
+
+- A directive can provide more context than an imperative.
+- You can combine several directives in one instruction.
+- It's usually better to implement a series of steps using a sequence of directives. If you tell the model to output the result of each step, when a problem arises you can easily see which step caused it. Although you can tell the model exactly what steps to follow, you can also just tell the model to break the instruction into steps, and then output the result of each step. This approach is called [chain of thought prompting](chain-of-thought-prompting.md).
+
+Instructions are typically more effective when used with examples. However, when you use both in a prompt you should make sure that the instructions are either above or below the examples for best model performance.
+
+Sometimes GPT models don't follow an instruction the way you expect because it doesn't provide enough context. You can add more context to an instruction by including primary and supporting content. You can include these when you add an instruction, and can add or adjust them after you test your instruction's effect.
+
+The earlier GPT models that generate text or code follow any instructions that you include in a prompt. Newer GPT models that support chat-based apps follow instructions that are in a [system message](https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/advanced-prompt-engineering?pivots=programming-language-chat-completions#system-message) or a user message, but not in an assistant message.
+
+### Primary content
+
+Primary content is text you add to an instruction for the model to process as if it were user input. For example, if you add the instruction **"Summarize US Presidential accomplishments."**, you could add a list of accomplishments as primary content.
+
+Make sure you clearly separate primary content from the instructions that apply to it, such as by labeling it. Consider the following C# example.
+
+```csharp
+prompt= @$"Instructions: Summarize US Presidential accomplishments.
+Accomplishments: 'George Washington
+First president of the United States.
+First president to have been a military veteran.
+First president to be elected to a second term in office.
+First president to receive votes from every presidential elector in an election.
+First president to fill the entire body of the United States federal judges; including the Supreme Court.
+First president to be declared an honorary citizen of a foreign country, and an honorary citizen of France.
+John Adams ...'" //Text truncated;
+```
+
+### Supporting content
+
+Supporting content is text that an instruction uses as input but isn't the subject of the instruction. The instruction must refer to the supporting content. As with primary content, supporting content should be clearly distinct from the instruction it supports.
```

Comment:
I would consider rewording the use of "input" here similar to how you did for the primary content section. Just want to make it clear that the model takes an input, but the supporting content is just part of the prompt that provides additional context for the instruction.

Reply:
revised